starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/CrdUploader.java

```diff
@@ -0,0 +1,91 @@
+package io.javaoperatorsdk.operator.springboot.starter;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.function.UnaryOperator;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+import static java.util.Arrays.stream;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class CrdUploader {
+
+  private static final Logger log = LoggerFactory.getLogger(CrdUploader.class);
+
+  private static final int CRD_READY_WAIT = 2000;
+
+  private final CrdTransformer crdTransformer;
+  private final KubernetesClient kubernetesClient;
+  private final boolean applyOnStartup;
+  private final String crdSuffix;
+  private final String crdPath;
+
+  public CrdUploader(KubernetesClient kubernetesClient, List<CrdTransformer> transformers,
+      boolean applyOnStartup, String crdPath, String crdSuffix) {
+    this.crdTransformer = CrdTransformer.reduce(transformers);
+    this.kubernetesClient = kubernetesClient;
+    this.applyOnStartup = applyOnStartup;
+    this.crdSuffix = crdSuffix;
+    this.crdPath = crdPath;
+  }
+
+  public void upload() throws Exception {
```

Comment:
Here we could have also a name that kinda describes that it applies if requested.  `applyConditionally()` of `applyIfNeeded()` might be an option?

Reply:
I don't really think adding implementation details to the method name is a great idea, perhaps if it was a method parameter. But if we're doing this as a bean `@conditional` i think this isn't relevant

Reply:
yes, in that case is not, and than i't fine. 

> I don't really think adding implementation details to the method name is a great idea,

unless it is a fundamental side effect, I'm not fan either just also can help realize issues and prevent debugging :)